### PR TITLE
#495 - implemented RepoPermissions.FromSettings.permissions() method

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #495 
Implemented `RepoPermissions.FromSettings.permissions()` method to return repository users and permissions

### CODE REVIEW

File: src/main/java/com/artipie/RepoPermissions.java [hunk 1]
```diff
@@ -110,8 +118,37 @@ public FromSettings(final Settings settings) {
...
+         * Credentials as yaml.
+         * @param key Repo settings key
+         * @return Completion action with yaml
+         */
+        private CompletionStage<YamlMapping> yaml(final Key key) {
+            return new RxStorageWrapper(this.storage())
+                .value(key)
+                .to(ContentAs.YAML)
+                .to(SingleInterop.get())
+                .thenApply(yaml -> (YamlMapping) yaml);
```
Comment (olegmoz) [inline src/main/java/com/artipie/RepoPermissions.java:151]: @olenagerasimova I'd suggest to remove dynamic casting and change return type to `<? extends YamlMapping>` or specify type param for `to` method as `<YamlMapping>`

File: src/main/java/com/artipie/RepoPermissions.java [hunk 2]
```diff
@@ -110,8 +118,37 @@ public FromSettings(final Settings settings) {
...
+            return this.yaml(new Key.From(String.format("%s.yaml", repo))).thenApply(
+                yaml -> Optional.ofNullable(yaml.yamlMapping("repo").yamlMapping("permissions"))
+                .map(
+                    perms -> {
+                        final Map<String, List<String>> res = new HashMap<>();
+                        perms.keys().forEach(
+                            node -> res.put(
+                                node.asScalar().value(),
+                                perms.yamlSequence(node.asScalar().value()).values().stream()
+                                .map(item -> Scalar.class.cast(item).value())
```
Comment (olegmoz) [inline src/main/java/com/artipie/RepoPermissions.java:131]: @olenagerasimova dynamic casting via `Scalar.class.cast(item)` could be avoided by using `item.asScalar()` method

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see two code improvement suggestions

**Comment by olenagerasimova** [comment]
@olegmoz thanks, all fixed

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, merging

**Pull request merged**